util/chunk/pool.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chunk
+
+import (
+	"container/list"
+	"math/rand"
+	"sync"
+
+	"github.com/pingcap/tidb/types"
+)
+
+// Pool is the column pool.
+type Pool struct {
+	varLenColPool   *colPool
+	fixLenColPool4  *colPool
+	fixLenColPool8  *colPool
+	fixLenColPool16 *colPool
+	fixLenColPool40 *colPool
```

Comment:
how about:

```
const shardNum = 8
type Pool [5]colPool
type colPool struct {
     shards  [shardNum]colPoolShard
 	 elemLen int
}

```

to make pool header in continual memory, so alloc once and keep in cache

Reply:
or 

```
type Pool struct {
 	varLenColPool   colPool
 	fixLenColPool4  colPool
 	fixLenColPool8  colPool
 	fixLenColPool16 colPool
 	fixLenColPool40 colPool
```
emm..maybe not use pointer in here.

Reply:
Uh, what is your concern?

Reply:
@zz-jason no need cal &colPool{} 5 times in `NewPool`,  memory can allocate continuous block.